src/viking.js

```diff
@@ -1,14 +1,100 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+  attack() {
+    return this.strength;
+  }
+  receiveDamage(damage) {
+    this.health-=damage;
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor (name, health, strength) {
+    super(health, strength);
+    this.name = name;
+  }
+  receiveDamage(damage) {
+    this.health-=damage;
+    if (this.health > 0) {
+      return this.name + " has received " + damage + " points of damage";
+    }
+    else {
+      return this.name + " has died in act of combat";
+    }
+  }
+  battleCry() {
+    return "Odin Owns You All!";
+  }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+  constructor(health, strength) {
+    super(health, strength);
+    }
+  receiveDamage(damage) {
+    this.health-=damage;
+    if (this.health > 0) {
+      return "A Saxon has received " + damage + " points of damage";
+    }
+    else {
+      return "A Saxon has died in combat";
+    }  
+  }
+}
+
 
 // War
-class War {}
+class War {
+  vikingArmy = []; // nicht mit this.... ?
+  saxonArmy = [];
+
+  addViking(viking) {
+    this.vikingArmy.push(viking);
+  }
+  addSaxon(saxon) {
+    this.saxonArmy.push(saxon);
+  }
+  vikingAttack() {
+      let randomIndexSaxon = Math.floor(Math.random() * this.saxonArmy.length);
+      let randomIndexViking = Math.floor(Math.random() * this.vikingArmy.length);
+      
+      this.saxonArmy[randomIndexSaxon].receiveDamage(this.vikingArmy[randomIndexViking].strength);
+      if (this.saxonArmy[randomIndexSaxon].health<=0) {
+      this.saxonArmy.splice(randomIndexSaxon, 1);
+      return "A Saxon has died in combat";
+      }  
+  }
+  saxonAttack() {
+    let randomIndexSaxon = Math.floor(Math.random() * this.saxonArmy.length);
+    let randomIndexViking = Math.floor(Math.random() * this.vikingArmy.length);
+    
+    this.vikingArmy[randomIndexViking].receiveDamage(this.saxonArmy[randomIndexSaxon].strength);
+    if (this.vikingArmy[randomIndexViking].health<=0) {
+    this.vikingArmy.splice(randomIndexViking, 1);
+    return       //return this.vikingArmy[randomIndexViking].name + " has received " + damage + " points of damage";
+    ;
+    }  
```

Comment:
To pass this last test, rememeber that you can save any result of calling a function in a variable, so for example, line 77 returns a string already -> you defined it in Saxon class:

```
  receiveDamage(damage) {
    this.health-=damage;
    if (this.health > 0) {
      return "A Saxon has received " + damage + " points of damage";
    }
    else {
      return "A Saxon has died in combat";
    }  
  }
```

so all you want to do is save this in a new variable 
```
const result = this.vikingArmy[randomIndexViking].receiveDamage(this.saxonArmy[randomIndexSaxon].strength);
```

and return this result at the end of the saxonAttack() method.
In other words calling saxonAttack() will return you a string